### DISCUSSION TITLE
Global concurrency limits - High CPU usage

### DISCUSSION THREAD

**Discussion by akharlamov** [dagster-io/dagster#19006] [General]
Hello, 

I am backfilling a considerable amount of partitions ~300 on our dataset. There are 3 different assets (gpt, bge, e5) building embeddings using the respective models. I want to run only 1 operation for each model concurrently, as every operation uses the full capacity of the respective service. 

I set dagster/concurrency_key to the appropriate limits, but I have one of the issues:
1. If I set max_concurrent_runs to the number of ops that can run in parallel (3 by the number of models), I see a single op hogging all available run slots. As only 1 gpt operation is allowed to run concurrently, the other two are just blocked (spin-locked???). That prevents other operations that can run in parallel (bge or e5) from running concurrently. 
2. I tried to remove the limit on the number of concurrent runs. But the blocked jobs seem to run in a tight loop and consume all available CPU instead of periodical checking. That leads to very high CPU load and causes a spectrum of overload related issues. 

Any suggestions?

**Comment by smackesey** [comment]
Hi @akharlamov, I'd like to clarify your setup in terms of Dagster abstractions:

- Are you using ops or just assets?
- Are you running asset jobs or just triggering asset backfills?

Would also be helpful if you could lay out a pseudo-code skeleton of your definitions.

**Comment by akharlamov** [comment]
@smackesey 
I use assets and trigger asset backfill.
